server/selection_test.go

```diff
@@ -355,3 +355,48 @@ func TestMinLSSelector_SelectUnknownSession_NilStakeReader(t *testing.T) {
 		i++
 	}
 }
+
+func TestMinLSSelector_RemoveUnknownSession(t *testing.T) {
+	assert := assert.New(t)
+
+	sel := NewMinLSSelector(nil, 1.0)
+
+	// Use ManifestID to identify each session
+	sessions := []*BroadcastSession{
+		&BroadcastSession{ManifestID: "foo"},
+		&BroadcastSession{ManifestID: "bar"},
+		&BroadcastSession{ManifestID: "baz"},
+	}
+
+	resetUnknownSessions := func() {
+		// Make a copy of the original slice so we can reset unknownSessions to the original slice
+		sel.unknownSessions = make([]*BroadcastSession, len(sessions))
+		copy(sel.unknownSessions, sessions)
+	}
+
+	// Test remove from front of list
+	resetUnknownSessions()
+	sel.removeUnknownSession(0)
+	assert.Len(sel.unknownSessions, 2)
+	assert.Equal("baz", string(sel.unknownSessions[0].ManifestID))
+	assert.Equal("bar", string(sel.unknownSessions[1].ManifestID))
+
+	// Test remove from middle of list
+	resetUnknownSessions()
```

Comment:
rather than reseting in the middle to initialize the next test, I would write four separate test functions, whose name indicates the specific aspect being tested

Reply:
Lately I've only been using multiple test functions with names that follow the convention you described when testing larger functions. In this case, given that the function under test is fairly small, I decided to keep all the test cases together instead of scattering them across different test functions. I personally find this easier to read, but this might just be matter of personal preference. If you feel strongly about it I can make the change.

Reply:
I think its ok, since the function is so small and there aren't many moving parts